### ISSUE TITLE
Implement unspecified websocket close code

### ISSUE THREAD

**Issue by jdm** [servo/servo#8158]
`tests/wpt/web-platform-tests/websockets/Secure-Close-1005-verify-code.htm` is a test that closes the websocket connection without a specified code (ie. the optional first argument is missing). Our implementation chooses not to send a status code in this case, but [the specification](https://tools.ietf.org/html/rfc6455#section-7.1.5) says that we should default to 1005 in this case in `WebSocket::Close`. This change should make at least one test pass when running `./mach test-wpt test/wpt/web-platform-tests/websockets`, so the steps for "Updating test expectations" in [the README](http://mxr.mozilla.org/servo/source/tests/wpt/README.md) should be followed.

Code: components/script/dom/websocket.rs

**Comment by yanirs** [comment]
Hey, having a look at this one. Is it simply a matter of adding an `else` clause on https://github.com/servo/servo/blob/master/components/script/dom/websocket.rs#L455? It seems to fix the issue.

Also, are there constants defined for these codes somewhere? Setting the code to 1005 doesn't feel right, but this seems to be the convention elsewhere in the file, e.g., on https://github.com/servo/servo/blob/master/components/script/dom/websocket.rs#L430

**Comment by jdm** [comment]
It wouldn't hurt to define some named constants for the values in https://tools.ietf.org/html/rfc6455#section-7.4.1 . And yes, an else block or changing it to use unwrap_or would make sense.

**Comment by yanirs** [comment]
Thanks! What's the idiomatic way of defining such constants in Rust?
I tried an `enum` as in [net_traits/net_error_list.rs](https://github.com/servo/servo/blob/master/components/net_traits/net_error_list.rs), but this yields an error when comparing to `u16`. Defining global `u16` constants yields dead code warnings for the ones that aren't used, so I figured they could be put in their own module where the dead code warning is suppressed. Alternatively, I could define constants just for the codes that are currently used. Which option is preferable in this case?
